### PR TITLE
Implement docker support for maven build

### PR DESCRIPTION
**Pull request by longshorej**
```bash
mvn clean package docker:build
sandbox run 2.1.0-alpha.1

echo "* loading activity-stream"
ACTIVITY_STREAM_ID="$(docker save chirper/activity-stream-impl | bndl --endpoint activity-stream --endpoint akka-remote | conduct load -q)"

echo "* loading chirp"
CHIRP_ID="$(docker save chirper/chirp-impl | bndl --endpoint chirp --endpoint akka-remote | conduct load -q)"

echo "* loading friend"
FRIEND_ID="$(docker save chirper/friend-impl | bndl --endpoint friend --endpoint akka-remote | conduct load -q)"

echo "* loading front-end"
FRONT_END_ID="$(docker save chirper/front-end | bndl --endpoint front-end --endpoint akka-remote | conduct load -q)"

echo "* running activity-stream"
conduct run "$ACTIVITY_STREAM_ID"
echo "* running chirp"
conduct run "$CHIRP_ID"
echo "* running friend"
conduct run "$FRIEND_ID"
echo "* running front-end"
conduct run "$FRONT_END_ID"

```

### CODE REVIEW

**Review by huntc — APPROVED** [review]
I'm really liking this...

One consideration: Maven plugins permit configuration to be inherited. You may be able to cut down on the boilerplate for the `docker-maven-plugin` by declaring common configuration in the parent pom. Some of the existing plugins already do that.

**Comment by markusjura** [comment]
@longshorej Looking good. Note that due to the change in https://github.com/typesafehub/conductr-cli/pull/443 the `--component` argument does not need to be specified anymore.

File: DEPLOY.md [hunk 1]
```diff
@@ -51,15 +51,16 @@ If you haven't had [SBT](http://www.scala-sbt.org/download.html) installed, proc
 
 _If you are using Windows, please ensure `sbt` is installed on the Linux VM._
 
-### Docker
+### Maven
+[Maven](https://maven.apache.org/) is an another build tool that Lagom Chirper is configured for. If you wish to try out the Maven support, make sure that you've [installed it](https://maven.apache.org/download.cgi).
```
Comment (edwardcallahan) [inline DEPLOY.md]: Note Docker requirement in Maven section please

**Review by edwardcallahan — CHANGES_REQUESTED** [review]
Review of Deploy.md
 The product name is and some of the early section is again outdated.
 Maven stuff looks good. 
  Should note Docker prereq in Maven section. 
  Consider maven as the default over sbt

**Review by edwardcallahan — APPROVED** [review]

**Review by huntc — APPROVED** [review]

**Pull request merged**